src/guc.c

```diff
@@ -360,15 +360,15 @@ _guc_init(void)
 							 NULL,
 							 NULL);
 
-	DefineCustomStringVariable(/* name= */ "timescaledb.license_key",
-							   /* short_dec= */ "TimescaleDB license key",
+	DefineCustomStringVariable(/* name= */ "timescaledb.license",
+							   /* short_dec= */ "TimescaleDB license type",
 							   /* long_dec= */ "Determines which features are enabled",
-							   /* valueAddr= */ &ts_guc_license_key,
-							   /* bootValue= */ TS_DEFAULT_LICENSE,
+							   /* valueAddr= */ &ts_guc_license,
+							   /* bootValue= */ TS_LICENSE_DEFAULT,
 							   /* context= */ PGC_SUSET,
-							   /* flags= */ GUC_SUPERUSER_ONLY,
-							   /* check_hook= */ ts_license_update_check,
-							   /* assign_hook= */ ts_license_on_assign,
+							   /* flags= */ 0,
```

Comment:
Shouldn't this still be superuser only?

Reply:
I'll double check that, but I think it is still is because of the used context. If am not mistaken, GUC_SUPERUSER_ONLY means to make this guc visible to superuser only, which makes sense because previously it could have an enterprise license key, which we did not want to expose.

Reply:
Yeah, I think you are right. Looks like this is only to restrict who can see the GUC. Do we have a check, though, that no-one else but the super user can set the GUC? I am thinking of the "upgrade" case, where a superuser at runtime can "upgrade" to TSL (but not downgrade to Apache).

Reply:
Yes. We have a permission test case